classes/local/util/manager.php

```diff
@@ -264,6 +268,28 @@ protected function prepare_observers() {
         }
     }
 
+    /*
+     * Prepare the event class files.
+     */
+    protected function prepare_events() {
+
+        foreach ($this->recipe['events'] as $event) {
+            if (empty($event['eventname'])) {
+                throw new exception('Missing event name');
+            }
+
+            $eventrecipe = $this->recipe;
+            $eventrecipe['event'] = $event;
```

Comment:
Maybe we could for purposes like this (dynamically computed recipe values) consistently use the `$recipe['self']` area and stick to it. So inside the skel class, it would be like $this->data['self']['event'].